force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -212,10 +260,16 @@ public with sharing class ClassesObjectsMethods {
      * @return The a positive number years between the two dates, or null if either date is null.
      */
     public static Integer calculateYearDifference(Date d1, Date d2) {
-
+        //Check for null dates
+        if (d1 == null || d2 == null) {
+            return null;
+        }
         // Subtract the difference in years
+        Integer firstYear = d1.year();
+        Integer secondYear = d2.year(); 
+        Integer differenceInYears = Math.abs(secondYear - firstYear);
 
-        return null; // Replace null with the variable you used to store the result
+        return differenceInYears; 
```

Comment:
Good job. I always encourage new devs to think if they really need those variables. If all I am doing with data is returning it. I try to return directly and skip creating variables unless I think it aids in readability. 

```
 public static Integer calculateYearDifference(Date d1, Date d2) {
        if (d1 == null || d2 == null) {
            return null;
        }
        return Math.abs(d1.year() - d2.year());
    }
 ```